File: Leetcode/linkedlist/19_Remove Nth Node from End of List/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int[] N = {n};
        head = recursiveFindAndRemove(head, N);
        return head;
    }
        private ListNode recursiveFindAndRemove(ListNode head, int[] N){
            if(head==null) return head;
            head.next = recursiveFindAndRemove(head.next, N);
            N[0]--;
            return (N[0]==0)? head.next: head;
        }
}

File: Leetcode/linkedlist/19_Remove Nth Node from End of List/Solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        N = [n]
        head = self.recursiveFindAndRemove(head, N)
        return head

    def recursiveFindAndRemove(self, head, N):
        if not head: return head
        head.next = self.recursiveFindAndRemove(head.next, N)
        N[0]-=1
        return head.next if N[0]==0 else head
